packages/browser-wallet/src/shared/utils/number-helpers.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Given a list of numbers return the smallest unused non-negative integer.
+ */
+export function getNextUnused(used: number[]) {
+    const sorted = used.sort((a, b) => a - b);
+    // If there are no credentials or 0 is not used, use 0
+    if (!sorted.length || sorted[0] > 0) {
+        return 0;
+    }
+    let i = 0;
+    while (i + 1 < sorted.length) {
+        // If the next credNumber is not the succesor of the current, there is a hole that can be filled
+        if (sorted[i] + 1 < sorted[i + 1]) {
+            break;
```

Comment:
With a for loop you could just return directly with the value here.

Reply:
Ah, sure, you would still need a return value outside of the loop in case the condition was never satisfied. I guess it's pretty much equal either way. Let's just keep it as is.